packages/jellyfish-api-core/__tests__/category/governance/createCfp.test.ts

```diff
@@ -0,0 +1,146 @@
+  })
+})
+
+describe('Governance', () => {
+  const container = new MasterNodeRegTestContainer(undefined, 'defi/defichain:HEAD-a61f615')
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should createCfp', async () => {
+    const data = {
+      title: 'Testing new community fund proposal',
+      amount: new BigNumber(100),
+      cycles: 2,
+      payoutAddress: await container.call('getnewaddress')
+    }
+    const proposalTx = await client.governance.createCfp(data)
+    await container.generate(1)
+
+    const proposal = await container.call('getproposal', [proposalTx])
+    expect(proposal.title).toStrictEqual(data.title)
+    expect(proposal.type).toStrictEqual(ProposalType.COMMUNITY_FUND_REQUEST)
+    expect(proposal.status).toStrictEqual(ProposalStatus.VOTING)
+    expect(proposal.amount).toStrictEqual(data.amount.toNumber())
+    expect(proposal.cyclesPaid).toStrictEqual(1)
+    expect(proposal.totalCycles).toStrictEqual(data.cycles)
+    expect(proposal.payoutAddress).toStrictEqual(data.payoutAddress)
+  })
+
+  it('should createCfp with inputs', async () => {
+    const data = {
+      title: 'Testing new community fund proposal',
+      amount: new BigNumber(100),
+      cycles: 2,
+      payoutAddress: await container.call('getnewaddress')
+    }
+    const utxos = await container.call('listunspent')
+    const inputs: UTXO[] = [
+      { txid: utxos[0].txid, vout: utxos[0].vout },
+      { txid: utxos[1].txid, vout: utxos[1].vout }
+    ]
+    const proposalTx = await client.governance.createCfp(data, inputs)
+    await container.generate(1)
+
+    expect(typeof proposalTx).toStrictEqual('string')
+  })
+
+  it('should not createCfp with cycles < 1', async () => {
+    const data = {
+      title: 'Testing another community fund proposal',
+      amount: new BigNumber(100),
+      payoutAddress: await container.call('getnewaddress'),
+      cycles: 0
+    }
+    const promise = client.governance.createCfp(data)
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow("RpcApiError: '<cycles> should be between 1 and 3', code: -8, method: createcfp")
+  })
+
+  it('should not createCfp with cycle > 3', async () => {
+    const data = {
+      title: 'Testing another community fund proposal',
+      amount: new BigNumber(100),
+      payoutAddress: await container.call('getnewaddress'),
+      cycles: 4
+    }
+    const promise = client.governance.createCfp(data)
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow("RpcApiError: '<cycles> should be between 1 and 3', code: -8, method: createcfp")
+  })
+
+  it('should not createCfp with unknown address type', async () => {
```

Comment:
Rearrange in this order

1. should createCfp
2. should not createCfp with cycles < 1
3. should not createCfp with cycle > 3
4. should not createCfp with unknown address type
5. should createCfp with with utxos
6. should not createCfp with arbitrary utxos (Add this too)